File: modules/core/src/effects/lighting/camera-light.js
import {PointLight} from './point-light';
import {getUniformsFromViewport} from '../../shaderlib/project/viewport-uniforms';

export default class CameraLight extends PointLight {
  getProjectedLight({layer}) {
    const {projectedLight} = this;
    const viewport = layer.context.viewport;
    const {coordinateSystem, coordinateOrigin, modelMatrix} = layer.props;
    // @ts-expect-error
    const {project_uCameraPosition} = getUniformsFromViewport({
      viewport,
      modelMatrix,
      coordinateSystem,
      coordinateOrigin
    });
    projectedLight.color = this.color;
    projectedLight.intensity = this.intensity;
    projectedLight.position = project_uCameraPosition;
    return projectedLight;
  }
}

File: bindings/pydeck/examples/lighting/camera_light.py
"""CameraLight
=============

Attach a point light to the current camera position.
"""

import pydeck as pdk

from _utils import write_lighting_example

write_lighting_example(
    pdk.Effect("CameraLight", color=[210, 230, 255], intensity=2.5),
    "camera_light.html",
)
